Test/Integration/Model/ScriptGeneratorTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CustomGento\Cookiebot\Test\Integration\Model;
+
+use CustomGento\Cookiebot\Model\Config;
+use CustomGento\Cookiebot\Model\ScriptGenerator;
+use Magento\Framework\App\ObjectManager;
+use Magento\TestFramework\ObjectManager as TestFrameworkObjectManager;
+use PHPUnit\Framework\TestCase;
+
+class ScriptGeneratorTest extends TestCase
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->config = TestFrameworkObjectManager::getInstance()->create(Config::class);
+    }
+
+    public function testEuCdnEnabledReturnsFalseByDefault(): void
+    {
+        self::assertFalse($this->config->isEuCdn());
+    }
+
+    /**
+     * @magentoConfigFixture default/web/cookiebot/enabled_eu_cdn 0
+     */
+    public function testIsEuCdndReturnsFalse(): void
```

Comment:
Good tests :+1: Feedback applies to this and the next test:

1. Please improve the naming. This does not test if `IsEuCdndReturnsFalse`. I think something like `testEuCdnIsUsedWhenDisabled` would be much more descriptive.
2. I think `AddScriptTest` should be extended by the EU option (each test can be run with EU enabled / disabled), so that it also tests if the right script is actually inserted on the page.
3. I don't think this test would currently pass as `%s` is replaced.

Reply:
It's fixed, my tests are completely ok, but old tests return some errors, and we can debug them If you think in this task is a good place for fixing them please let me know

Reply:
All tests should pass :)